devicetypes/smartthings/vision-in-wall-2relays-switch.src/vision-in-wall-2relays-switch.groovy

```diff
@@ -0,0 +1,277 @@
+/**
+ *  Vision In-Wall 2Relays Switch (Models: ZL7435xx-5)
+ * 
+ *  Author:
+ *    Lan, Kuo Wei
+ *
+ *	Product Link:
+ *    http://www.visionsecurity.com.tw/index.php?option=product&lang=en&task=pageinfo&id=335&belongid=334&index=0
+*/
+metadata {
+	definition(
+		name: "Vision In-Wall 2Relays Switch", 
+		namespace: "SmartThings", 
+		author: "Lan, Kuo Wei", 
+		ocfDeviceType: "oic.d.switch", 
+		mnmn: "SmartThings", 
+		vid: "generic-switch"
+	) {
+		capability "Actuator"
+		capability "Health Check"
+		capability "Refresh"
+		capability "Switch"
+		capability "Configuration"
+		// This DTH uses 2 switch endpoints. Parent DTH controls endpoint 1 so please use '1' at the end of deviceJoinName
+		// Child device (isComponent : false) representing endpoint 2 will substitute 1 with 2 for easier identification.
+		fingerprint manufacturer: "0109", prod: "2017", model: "171B", deviceJoinName: "Vision 2Relays Switch 1" //zw:Ls type:1001 mfr:0109 prod:2017 model:171B ver:16.11 zwv:4.38 lib:03 cc:98 sec:5E,72,86,85,59,70,5A,7A,60,8E,73,27,25 epc:2
+		fingerprint manufacturer: "0109", prod: "2017", model: "171C", deviceJoinName: "Vision 2Relays Switch 1" //zw:Ls type:1001 mfr:0109 prod:2017 model:171C ver:25.07 zwv:4.54 lib:03 cc:98 sec:5E,72,86,85,59,70,5A,7A,60,8E,73,27,25 epc:2
+	}
+
+	preferences {
+		parameterMap.each {
+			input (title: it.name, description: it.description, type: "paragraph", element: "paragraph")
+			switch(it.type) {
+				case "boolean":
+					input(type: "paragraph", element: "paragraph", description: "Option enabled: ${it.activeDescription}\n" +
+							"Option disabled: ${it.inactiveDescription}"
+					)
+					input(name: it.key, type: "boolean", title: "Enable", defaultValue: it.defaultValue == it.activeOption, required: false)
+					break
+					
+				case "enum":
+					input(name: it.key, title: "Select", type: "enum", options: it.values, defaultValue: it.defaultValue, required: false)
+					break
+			}	
+		}
+	}
+}
+
+def installed() {
+	def componentLabel
+	if (device.displayName.endsWith('1')) {
+		componentLabel = "${device.displayName[0..-2]}2"
+	} else {		
+		componentLabel = "$device.displayName 2"
+	}
+	try {
+		String dni = "${device.deviceNetworkId}-ep2"
```

Comment:
the convention is to just use `:2` for child device DNIs

Reply:
Yes, I will fix it,
please wait me fix namespace's problem,
that have some issue need test.

Reply:
Already fix on 2021/02/22.